### PR TITLE
[3.4] fix create collection timeouts test agent

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3848/

File: arangod/Cluster/SynchronizeShard.cpp [hunk 1]
```diff
@@ -384,89 +384,117 @@ static inline bool isStopping() {
...
-    if (result != nullptr && result->getHttpReturnCode() == 200) {
-      auto const vp = putres->result->getBodyVelocyPack();
-      auto const& slice = vp->slice();
-      TRI_ASSERT(slice.isObject());
-      if (slice.hasKey("lockHeld") && slice.get("lockHeld").isBoolean() &&
-          slice.get("lockHeld").getBool()) {
+    // See if above POST has been returned (>= 3.4 only)
+    auto enqres = cc->enquire(postres);
+    if (enqres.status == CL_COMM_SENT) {
+      if (enqres.result != nullptr && enqres.result->getHttpReturnCode() == 200) {
```
Comment (neunhoef) [inline arangod/Cluster/SynchronizeShard.cpp]: Need to call `cc->drop(postres)` in this case, otherwise the structure remains indefinitely in `ClusterComm` data structures.

File: arangod/Cluster/SynchronizeShard.cpp [hunk 2]
```diff
@@ -384,89 +384,117 @@ static inline bool isStopping() {
...
+        TRI_ASSERT(slice.isObject());
+        if (slice.hasKey("lockHeld") && slice.get("lockHeld").isBoolean() &&
+            slice.get("lockHeld").getBool()) {    // Habemus clausum
+          return arangodb::Result();
+        }
+        LOG_TOPIC(DEBUG, Logger::MAINTENANCE)
+          << "startReadLockOnLeader: Lock not yet acquired, retrying... ";
+      } else if (enqres.status == CL_COMM_SENT) { // POST sent in meantime: Error (>=3.4)
+        LOG_TOPIC(DEBUG, Logger::MAINTENANCE)
+          << "startReadLockOnLeader: Failed to acquire read lock: "
```
Comment (neunhoef) [inline arangod/Cluster/SynchronizeShard.cpp]: Should we not bail out and give up here? Further waiting does not make sense.

File: tests/js/server/resilience/moving-shards-cluster.js [hunk 3]
```diff
@@ -35,11 +35,15 @@ const wait = require("internal").wait;
 const supervisionState = require("@arangodb/cluster").supervisionState;
 
 function getDBServers() {
-  var tmp = global.ArangoClusterInfo.getDBServers();
-  var servers = [];
-  for (var i = 0; i < tmp.length; ++i) {
-    servers[i] = tmp[i].serverId;
+  var tmp, servers = [];
```
Comment (neunhoef) [inline tests/js/server/resilience/moving-shards-cluster.js]: Let's remove this "fix" here and make a separate PR to fix the global variable `servers` in all branches.

File: arangod/Cluster/SynchronizeShard.cpp (hunk 2 above)
Comment (kvahed) [reply arangod/Cluster/SynchronizeShard.cpp]: done.

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3868/
